editor/scene_tree_dock.h

```diff
@@ -247,6 +247,9 @@ class SceneTreeDock : public VBoxContainer {
 	static SceneTreeDock *singleton;
 	static void _update_configuration_warning();
 
+	static bool _update_node_path(const NodePath &p_root_path, NodePath &p_node_path, List<Pair<NodePath, NodePath>> *p_renames);
+	static bool _check_node_path_recursive(const NodePath &p_root_path, Variant &p_variant, List<Pair<NodePath, NodePath>> *p_renames);
```

Comment:
I was just wondering, do those actually need to be `static`?

Also, since `p_node_path` and `p_variant` are used to return new values, they should probably be named with a `r_` prefix instead (our convention for a mutable reference that will get changed by the method).

Reply:
They don't have to be `static` but they can. I generally set methods as `static` when possible to make it clear they don't use anything from the class. Do we have a different rule about when to use `static`?

Makes sense for the missing `_r`, prefix, I can make a fixup PR to change it.

Reply:
> They don't have to be `static` but they can. I generally set methods as `static` when possible to make it clear they don't use anything from the class. Do we have a different rule about when to use `static`?

I don't think we have a specific rule, I'm just asking out of curiosity (and lack of expertise on the actual impact of making things static in C++). If it doesn't hurt and makes the code clearer, that's fine with me :)